bindings/gumjs/gumv8core.cpp

```diff
@@ -2276,52 +2284,90 @@ gum_v8_native_function_invoke (GumV8NativeFunction * self,
 {
   auto core = (GumV8Core *) info.Data ().As<External> ()->Value ();
   auto isolate = core->isolate;
-  gsize num_args_required = self->cif.nargs;
+  auto cif = &self->cif;
+  gsize num_args_declared = cif->nargs;
   gsize num_args_provided = (argv != nullptr) ? argc : info.Length ();
+  gsize num_args_fixed = self->nargs_fixed;
+  gboolean is_variadic = self->is_variadic;
 
-  if (num_args_provided != num_args_required)
+  if ((is_variadic && num_args_provided < num_args_fixed)
+   || (!is_variadic && num_args_provided != num_args_declared))
   {
     _gum_v8_throw_ascii_literal (isolate, "bad argument count");
     return;
   }
 
-  auto rtype = self->cif.rtype;
-
+  auto rtype = cif->rtype;
+  auto atypes = cif->arg_types;
   gsize rsize = MAX (rtype->size, sizeof (gsize));
   gsize ralign = MAX (rtype->alignment, sizeof (gsize));
   auto rvalue = (GumFFIValue *) g_alloca (rsize + ralign - 1);
   rvalue = GUM_ALIGN_POINTER (GumFFIValue *, rvalue, ralign);
 
   void ** avalue;
   guint8 * avalues;
+  ffi_cif tmp_cif;
+  GumFFIValue tmp_value = { 0, };
 
-  if (num_args_required > 0)
+  if (num_args_provided > 0)
   {
-    avalue = (void **) g_alloca (num_args_required * sizeof (void *));
+    gsize avalue_cnt = MAX (num_args_declared, num_args_provided);
+    avalue = (void **) g_alloca (avalue_cnt * sizeof (void *));
+
+    gsize arglist_size = self->arglist_size;
+    if (is_variadic && num_args_provided > num_args_declared)
+    {
+      atypes = g_newa (ffi_type *, num_args_provided);
+
+      memcpy (atypes, cif->arg_types, num_args_declared * sizeof (void *));
+      for (gsize i = num_args_declared, type_idx = num_args_fixed;
+           i != num_args_provided; i++)
+      {
+        ffi_type * t = cif->arg_types[type_idx];
+
+        atypes[i] = t;
+        arglist_size = GUM_ALIGN_SIZE (arglist_size, t->alignment);
+        arglist_size += t->size;
 
-    gsize arglist_alignment = self->cif.arg_types[0]->alignment;
-    avalues = (guint8 *) g_alloca (self->arglist_size + arglist_alignment - 1);
+        if (++type_idx >= num_args_declared)
+          type_idx = num_args_fixed;
+      }
+
+      cif = &tmp_cif;
+      if (ffi_prep_cif_var (cif, self->abi, num_args_fixed, num_args_provided,
+          rtype, atypes) != FFI_OK)
+      {
+        _gum_v8_throw_ascii_literal (isolate,
+           "failed to compile function call interface");
```

Comment:
```suggestion
            "failed to compile function call interface");
```

(Looks like the hanging indent is 3 spaces instead of 4?)